wweval/R/combine_and_summarize_ww_data.R

```diff
@@ -0,0 +1,368 @@
+#' Combine and summarize wastewater data
+#'
+#' This function iterates through the vector of forecast dates and locations,
+#' checks for the presence of wastewater data, and if present loads it in.
+#' It then computes a number of summary metrics on the wastewater data for that
+#' particular location and forecast date to produce a single row of output
+#' metadata. Locations without wastewater data are indicated as such and have
+#' NAs for the wastewater specific entries
+#'
+#' @param forecast_dates The vector of character strings of all the forecast
+#' dates
+#' @param locations The vector of character strings of all the locations
+#' @param eval_output_subdir The outer subdirectory of the nested file structure
+#'
+#' @return ww_metadata: a tibble with a row for each forecast date location and
+#' columns that provide summaries of the wastewater data
+#' @export
+combine_and_summarize_ww_data <- function(forecast_dates,
+                                          locations,
+                                          eval_output_subdir) {
+  if (length(forecast_dates) != length(locations)) {
+    cli::cli_abort(
+      message = "Vector of forecast dates and locations must be equal in length"
+    )
+  }
+
+  ww_metadata <- tibble()
+  flag_failed_output <- tibble()
+
+  for (i in seq_along(forecast_dates)) {
+    this_forecast_date <- forecast_dates[i]
+    this_location <- locations[i]
+
+    fp <- get_filepath(
+      eval_output_subdir,
+      scenario = "status_quo",
+      forecast_date = this_forecast_date,
+      model_type = "ww",
+      location = this_location,
+      output_type = glue::glue("input_ww_data"),
+      file_extension = "tsv"
+    )
+    fp_hosp <- get_filepath(
+      eval_output_subdir,
+      scenario = "status_quo",
+      forecast_date = this_forecast_date,
+      model_type = "ww",
+      location = this_location,
+      output_type = glue::glue("input_hosp_data"),
+      file_extension = "tsv"
+    )
+
+    if (file.exists(fp)) {
+      tryCatch(
```

Comment:
The long `tryCatch` here suggests to me this should be broken into a helper function that is called iteratively and fails gracefully (e.g. via wrapping in `purrr::safely`)